code/modules/food&drinks/food/snacks_egg.dm

```diff
@@ -18,8 +18,9 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/egg/throw_impact(atom/hit_atom)
 	..()
-	new/obj/effect/decal/cleanable/egg_smudge(src.loc)
 	reagents.reaction(hit_atom, TOUCH)
+	var/obj/effect/decal/cleanable/eggdecal = new/obj/effect/decal/cleanable/egg_smudge(src.loc)
```

Comment:
I'd recommend that the transfer is made to not be 100% efficient as the decal will still be some left on the floor when scooping it up.